suite-native/module-send/src/components/CryptoAmountInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { ReactNode, RefObject } from 'react';
+import Animated, { SharedValue, useAnimatedStyle } from 'react-native-reanimated';
+import { TextInput } from 'react-native';
+
+import { Text, Input } from '@suite-native/atoms';
+import { useFormContext } from '@suite-native/forms';
+import { useField } from '@suite-native/forms';
+import { useCryptoFiatConverters } from '@suite-native/formatters';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useFormatters } from '@suite-common/formatters';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { Color } from '@trezor/theme';
+import { useDebounce } from '@trezor/react-utils';
+
+export type SendAmountInputProps = {
+    recipientIndex: number;
+    networkSymbol: NetworkSymbol;
+    inputRef: RefObject<TextInput>;
+    scaleValue: SharedValue<number>;
+    translateValue: SharedValue<number>;
+    isDisabled?: boolean;
+};
+
+export const sendAmountInputWrapperStyle = prepareNativeStyle<{ isDisabled: boolean }>(
+    (_, { isDisabled }) => ({
+        position: 'absolute',
+        top: isDisabled ? 2 : 0,
+        flex: 1,
+        width: '100%',
+    }),
+);
+
+export const SendAmountCurrencyLabelWrapper = ({
+    children,
+    isDisabled,
+}: {
+    children: ReactNode;
+    isDisabled: boolean;
+}) => {
+    const textColor: Color = isDisabled ? 'textSubdued' : 'textDefault';
+
+    return <Text color={textColor}>{children}</Text>;
+};
+
+export const sendAmountTransformer = (value: string) =>
```

Comment:
Please, put this and everything else, that is reused in FiatAmountInput, to other file (utils?)

Reply:
I moved the transformer and also type declaration to utils and types files. I kept the style here, because IMO the styles should always stay next to the JSX component. We have this patter of reusing styles to other components implemented also in other files of the codebase, so I think it is OK. Let me know what you think.

264b35313b